Run  demo_genesis.sh failed.

Steps,

```
git clone https://github.com/AMDResearch/Ryzers
cd Ryzers/
uv venv --python 3.12
uv pip install -e .
# Build
ryzers build genesis

# Launch the container
ryzers run bash
```


Failed with log,


```bash
root@GZ302EA:/ryzers# ./demo_genesis.sh
[Genesis] [04:30:46] [INFO] ╭─────────────────────────────────────────────────────────────────────────────────────╮
[Genesis] [04:30:46] [INFO] │┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉ Genesis ┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉┈┉│
[Genesis] [04:30:46] [INFO] ╰─────────────────────────────────────────────────────────────────────────────────────╯
[Genesis] [04:30:46] [INFO] Consider setting 'performance_mode=True' in production to maximise runtime speed, if significantly increasing compilation time is not a concern.
[Genesis] [04:30:46] [INFO] Running on [AMD Radeon Graphics] with backend gs.vulkan. Device memory: 60.75 GB.
[Genesis] [04:30:46] [INFO] 🚀 Genesis initialized. 🔖 version: 0.3.3, 🌱 seed: None, 📏 precision: '32', 🐛 debug: False, 🎨 theme: 'dark'.
MoTTY X11 proxy: Unsupported authorisation protocol

Traceback (most recent call last):
  File "/opt/venv/lib/python3.12/site-packages/genesis/vis/visualizer.py", line 58, in __init__
    display = pyglet.display.get_display()
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.12/site-packages/pyglet/display/__init__.py", line 77, in get_display
    return Display()
           ^^^^^^^^^
  File "/opt/venv/lib/python3.12/site-packages/pyglet/display/xlib.py", line 102, in __init__
    raise NoSuchDisplayException(f'Cannot connect to "{name}"')
pyglet.display.xlib.NoSuchDisplayException: Cannot connect to "None"

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/ryzers/Genesis/examples/tutorials/sph_liquid.py", line 11, in <module>
    scene = gs.Scene(
            ^^^^^^^^^
  File "/opt/venv/lib/python3.12/site-packages/genesis/utils/misc.py", line 126, in new_init
    original_init(self, *args, **kwargs)
  File "/opt/venv/lib/python3.12/site-packages/genesis/engine/scene.py", line 189, in __init__
    self._visualizer = Visualizer(
                       ^^^^^^^^^^^
  File "/opt/venv/lib/python3.12/site-packages/genesis/vis/visualizer.py", line 69, in __init__
    gs.raise_exception_from("No display detected. Use `show_viewer=False` for headless mode.", e)
  File "/opt/venv/lib/python3.12/site-packages/genesis/utils/misc.py", line 43, in raise_exception_from
    raise gs.GenesisException(msg) from cause
genesis.GenesisException: No display detected. Use `show_viewer=False` for headless mode.

[Genesis] [04:30:46] [ERROR] GenesisException: No display detected. Use `show_viewer=False` for headless mode.
[Genesis] [04:30:46] [INFO] 💤 Exiting Genesis and caching compiled kernels...
Exception ignored in: <function Scene.__del__ at 0x770fb5b894e0>
Traceback (most recent call last):
  File "/opt/venv/lib/python3.12/site-packages/genesis/engine/scene.py", line 210, in __del__
AttributeError: 'Scene' object has no attribute '_visualizer'
```

--- a/packages/robotics/genesis/Dockerfile
+++ b/packages/robotics/genesis/Dockerfile
@@ -14,12 +14,13 @@ ENV DEBIAN_FRONTEND=noninteractive \
     PYTHONUNBUFFERED=1 \
     HSA_OVERRIDE_GFX_VERSION=11.0.0
 
-# System packages: Vulkan + FFmpeg + minimal GUI dependencies
+# System packages: Vulkan + FFmpeg + EGL for headless rendering
 RUN apt-get update && apt-get install -y --no-install-recommends \
     curl wget git ca-certificates locales \
     libvulkan1 mesa-vulkan-drivers vulkan-tools \
     ffmpeg \
     libgl1 libglib2.0-0 \
+    libegl1-mesa-dev libgles2-mesa-dev \
  && rm -rf /var/lib/apt/lists/*
 
 # pip configuration (system-wide inside the container)

--- a/packages/robotics/genesis/config.yaml
+++ b/packages/robotics/genesis/config.yaml
@@ -15,8 +15,9 @@
 
 ### ryzers run options
 
-# environment_variables:              # List of environment variables to set in the container
-# - "ENV_VARIABLE=VALUE"              # E.g. "PYTHONPATH=/path/to/python"
+# Headless rendering support (fixes issue #66)
+environment_variables:
+  - "PYOPENGL_PLATFORM=egl"
 
 # port_mappings:  
 # - "host_portnum:container_portnum"  # List of port mappings to expose from the docker 

--- a/packages/robotics/genesis/test.py
+++ b/packages/robotics/genesis/test.py
@@ -4,13 +4,16 @@
 # SPDX-License-Identifier: MIT
 
 import os
-os.environ['PYOPENGL_PLATFORM'] = 'glx'
- 
+# Use EGL for headless rendering (issue #66)
+if 'PYOPENGL_PLATFORM' not in os.environ:
+    os.environ['PYOPENGL_PLATFORM'] = 'egl'
+
 import genesis as gs
 
 gs.init(backend=gs.vulkan)
 
-scene = gs.Scene(show_viewer=True)
+# Headless mode by default (set show_viewer=True if X11 display available)
+scene = gs.Scene(show_viewer=False)
 plane = scene.add_entity(gs.morphs.Plane())
 franka = scene.add_entity(
     gs.morphs.MJCF(file='xml/franka_emika_panda/panda.xml'),

--- a/packages/robotics/genesis/test.py
+++ b/packages/robotics/genesis/test.py
@@ -6,7 +6,9 @@
 import os
 # Use EGL for headless rendering (issue #66)
 if 'PYOPENGL_PLATFORM' not in os.environ:
-  os.environ['PYOPENGL_PLATFORM'] = 'glx'
+  os.environ['PYOPENGL_PLATFORM'] = 'egl'
+  
+import genesis as gs
 
 print("Testing Genesis installation...")
 

